lib/Views/src/layouts/DateTimeView32x8.cpp

```diff
@@ -68,20 +68,18 @@ void DateTimeView32x8::setCurrentTime(const tm& now)
 {
     /* update lamp widgets */
 
-    /* tm_wday starts at sunday, first lamp indicates monday.*/
-    uint8_t activeLamp = (0U < now.tm_wday) ? (now.tm_wday - 1U) : (MAX_LAMPS - 1U);
-
-    /* Last active lamp has to be deactivated. */
-    uint8_t lampToDeactivate = (0U < activeLamp) ? (activeLamp - 1U) : (MAX_LAMPS - 1U);
-
-    if (MAX_LAMPS > activeLamp)
+    /* tm_wday starts at sunday, first lamp specified via m_startOfWeek.*/
+    uint8_t activeLamp = now.tm_wday - m_startOfWeek;
+    if (MAX_LAMPS < activeLamp)
     {
-        m_lampWidgets[activeLamp].setOnState(true);
+        activeLamp += 7U;

```

Comment:
Use ```activeLamp %= MAX_LAMPS;``` which is generic.

Reply:
I've updated to use MAX_LAMPS, however, I've retained the conditional instead of modulo as this results in smaller code (at least on my board, esp32 dev kit v1) and should provide the same functionality.

Reply:
It doesn't work with "+=", because the following for loop index runs only in range of [0; MAX_LAMPS[. You need the modulo to keep it in this range.

Reply:
Are we talking about the same code? This is what I have in my latest commit:

```
     /* tm_wday starts at sunday, first lamp specified via m_startOfWeek.*/
     uint8_t activeLamp = now.tm_wday - m_startOfWeek;

     /* the above subtraction may underflow, so detect and correct for that.
      * this is more efficient than integer remainder on xtensa cores. */
     if (MAX_LAMPS < activeLamp)
     {
         activeLamp += MAX_LAMPS;
     }
 ```

`now.tm_wday` will be a value from 0 through 6, inclusive. `m_startOfWeek` will be 0 through 6 (`MAX_LAMPS - 1`), inclusive (thanks to the new checks). So there are two possibilities:

1. If `now.tm_wday >= m_startOfWeek`, then the result of the subtraction is also in the range of 0 through 6, inclusive. So at this point, we're done.
2. Otherwise, `m_startOfWeek` is greater than `now.tm_wday`, so the subtraction goes negative. But since this is an unsigned value, it actually goes positive, and will be in the range 250 through 255, inclusive. This will be greater than the conditional check, so a `+= MAX_LAMPS` will cause it to wrap around in the positive direction, and the resulting value will be  1 through 6.

Another way to write this same logic would be something like this:
```
uint8_t activeLamp = ((now.tm_wday >= m_startOfWeek) ? 0U : MAX_LAMPS) + now.tm_wday - m_startOfWeek;
```
however, the compiler doesn't seem to optimize this, and it also ends up a few bytes bigger.

Reply:
Yes, you right ... I shouldn't do different things in parallel, so I mixed it up. :-D